src/main/java/org/opendatasoft/elasticsearch/search/aggregations/bucket/PathHierarchyAggregationBuilder.java

```diff
@@ -320,16 +320,16 @@ protected XContentBuilder doXContentBody(XContentBuilder builder, Params params)
         return builder.endObject();
     }
 
-    /**
-     * Used for caching requests, amongst other things.
-     */
     @Override
     public int hashCode() {
-        return Objects.hash(separator, minDepth, maxDepth, depth, order, minDocCount, bucketCountThresholds);
+        return Objects.hash(super.hashCode(), separator, minDepth, maxDepth, depth, order, minDocCount, bucketCountThresholds);
     }
 
     @Override
     public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
```

Comment:
Why these two lines if it is already tested by the parent class ?

Reply:
yep I agree with you. I just followed the others ValuesSourceAggregationBuilder classes template.
For example the following is from `public class DateHistogramAggregationBuilder extends ValuesSourceAggregationBuilder<DateHistogramAggregationBuilder>`:

```java
    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), order, keyed, minDocCount, dateHistogramInterval, minDocCount, extendedBounds, hardBounds);
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        if (super.equals(obj) == false) return false;
        DateHistogramAggregationBuilder other = (DateHistogramAggregationBuilder) obj;
        return Objects.equals(order, other.order)
            && Objects.equals(keyed, other.keyed)
            && Objects.equals(minDocCount, other.minDocCount)
            && Objects.equals(dateHistogramInterval, other.dateHistogramInterval)
            && Objects.equals(offset, other.offset)
            && Objects.equals(extendedBounds, other.extendedBounds)
            && Objects.equals(hardBounds, other.hardBounds);
    }
```

Reply:
`getClass();` is legit because it will return PathHierarchy class while super call will return the parent class.